packages/dev/core/src/Physics/v2/physicsShape.ts

```diff
@@ -0,0 +1,239 @@
+import type { TransformNode } from "../../Meshes/transformNode";
+import { BoundingBox } from "../../Culling/boundingBox";
+import { ShapeType } from "./IPhysicsEnginePluginV2";
+import type { IPhysicsEnginePluginV2, PhysicsShapeParameters } from "./IPhysicsEnginePluginV2";
+import type { PhysicsMaterial } from "./physicsMaterial";
+import { Vector3 } from "../../Maths/math.vector";
+import type { Quaternion } from "../../Maths/math.vector";
+import type { AbstractMesh } from "../../Meshes/abstractMesh";
+import type { Scene } from "../../scene";
+
+/**
+ *
+ */
+export class PhysicsShape {
+    /** @internal */
+    public _pluginData: any = undefined;
+
+    private _physicsPlugin: IPhysicsEnginePluginV2;
+
+    private _type: ShapeType;
+
+    /**
+     *
+     * @param type
+     * @param options
+     * @param scene
+     * @returns
+     */
+    constructor(type: number, options: PhysicsShapeParameters = {}, scene: Scene) {
+        this._type = type;
+        if (!scene) {
+            return;
+        }
+
+        this._physicsPlugin = scene.getPhysicsEngine()?.getPhysicsPlugin() as any;
+        if (!this._physicsPlugin) {
+            return;

```

Comment:
throw here? and make the physics plugin not optional (and avoid using the `?` in every function)

Reply:
Fixed